### PR TITLE
feat(#210): add menu API and pull menu navigation using it

### PR DESCRIPTION
**Pull request by rafalkosla101**
added API to pull menus and refactored navigation to use pulled links
## Description

implemented API to pull menus from backend, added deserialisers to convert given data to proper format and changed navigation through categories to use Menu if possible, if not the menu is still being build based on item slugs

## Related Issue

#210
## Motivation and Context

it solves #210
## How Has This Been Tested?

manually
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/middleware.config.js [hunk 1]
```diff
@@ -24,6 +24,7 @@ module.exports = {
           // 'enabled' (Spree 4.4+)
           // 'legacy' (Spree older than 4.4 with spree_wishlist gem installed)
           // 'disabled' (Spree older than 4.4 without spree_wishlist gem)
+          useMenuApi: false,
```
Comment (rafalkosla101) [inline packages/theme/middleware.config.js:27]: I do not know if user should be able to set that property manually or it should be set automatically based on used spree version a few lines above. If it is set automatically there could be a few problems when moving from older spree to the newest ones(not properly set backend), but setting it manually may lead to some other issues. Any ideas?

File: packages/theme/pages/Category.vue [hunk 2]
```diff
@@ -259,11 +270,18 @@ export default {
       }
     };
 
+    onMounted(async () => {
+      try {
+        menu.value = await $spree.api.getMenus({menuType: 'header', menuName: 'Main menu', locale: locale.value});
```
Comment (rafalcymerys) [inline packages/theme/pages/Category.vue:275]: Could we maybe create a custom composable, e.g. `useMenus()` to load and cache menus between components?

File: packages/theme/components/AppFooter.vue [hunk 3]
```diff
@@ -0,0 +1,156 @@
+<template>
+  <SfFooter v-if="!menu.isDisabled" :column="parseInt(menuSize)" multiple class="footer">
```
[the remark this answers is not in the document]
Comment (rafalkosla101) [reply packages/theme/components/AppFooter.vue:2]: Sorry for not answering, if menu isDisabled default footer will be loaded which is declared a few lanes below with v-else statement

[the remark this answers is not in the document]
Comment (rafalcymerys) [reply packages/theme/components/AppFooter.vue:2]: My overlook, makes sense :)

**Pull request merged**